src/ui/components/filters/filteroptionscomponent.js

```diff
@@ -126,35 +126,25 @@ class FilterOptionsConfig {
     }
     // previousOptions will be null if there were no previousOptions in persistentStorage
     const previousOptions = config.previousOptions;
-    if (previousOptions) {
-      this.options = this.setPreviousOptions(this.options, previousOptions || []);
-    } else {
-      this.options = this.setSelected(this.options);
-    }
-  }
-
-  /**
-   * Sets selected options on load based on options stored in persistent storage.
-   * @param {Array<Object>} options
-   * @param {Array<string>} previousOptions
-   * @returns {Array<Object>}
-   */
-  setPreviousOptions (options, previousOptions) {
-    return options.map(o => ({
-      ...o,
-      selected: previousOptions.includes(o.label)
-    }));
+    this.options = this.setSelectedOptions(this.options, previousOptions);
   }
 
   /**
+   * Sets selected options on load based on options stored in persistent storage and options with selected: true.
    * If no previous options were stored in persistentStorage, default to options marked
    * as selected. If multiple options are marked as selected for 'singleoption', only the
    * first should be selected.
-   * @param {*} options
+   * @param {Array<Object>} options
+   * @param {Array<string>} previousOptions
    * @returns {Array<Object>}
    */
-  setSelected (options) {
-    if (this.control === 'singleoption') {
+  setSelectedOptions (options, previousOptions) {
+    if (previousOptions) {
```

Comment:
I think there's a weird edge case here. Someone could first configure FilterOptions to be 'multioption', then save multiple selected options in persistent storage. If they then decide to change FilterOptions to be 'singleoption', but use the URL from before, could they wind up with multiple options selected here, even though it's 'singleoption'?

Reply:
Ah I didn't consider that. What happens is that multiple radio buttons are marked as checked in the html. In Chrome and Firefox this is interpreted as visually only the last radio button being selected, however if you inspect the dom multiple radio buttons are given the 'checked' attribute. If you apply the filters, since the component doesn't care about whether it's multioption/singleoption when it builds the filter it includes all the selected options.

Is this something we need to handle? I think no matter what we do the url won't be valid anymore.

Reply:
So they could generate a filter with multiple options selected even though it's type 'singleoption'? I think we should handle this, even though it's an extreme edge case. For consistency's sake, I think we should just pick the first of the previousOptions in this case. What do you think?

Reply:
yeah sounds good to me! could only be a good thing to be accurate

Reply:
updated to add this handling, also added unit tests for RADIUS_FILTER and options.selected logic